internal/resources/cluster/nodepools/data_source_node_pool.go

```diff
@@ -7,35 +7,68 @@ package nodepools
 
 import (
 	"context"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/pkg/errors"
 
 	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
 	nodepoolsmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/nodepool"
 	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
 )
 
+type contextMethodKey struct{}
+
 func DataSourceClusterNodePool() *schema.Resource {
 	return &schema.Resource{
 		ReadContext: dataSourceClusterNodePoolRead,
 		Schema:      nodePoolSchema,
 	}
 }
 
-func dataSourceClusterNodePoolRead(_ context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+func dataSourceClusterNodePoolRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
 	config := m.(authctx.TanzuContext)
 
-	resp, err := config.TMCConnection.NodePoolResourceService.ManageV1alpha1ClusterNodePoolResourceServiceGet(constructFullName(d))
+	var (
+		resp *nodepoolsmodel.VmwareTanzuManageV1alpha1ClusterNodepoolCreateNodepoolResponse
+		err  error
+	)
+
+	getNodepoolResourceRetryableFn := func() (retry bool, err error) {
+		resp, err := config.TMCConnection.NodePoolResourceService.ManageV1alpha1ClusterNodePoolResourceServiceGet(constructFullName(d))
+		if err != nil {
+			if clienterrors.IsNotFoundError(err) {
+				return true, nil
+			}
+
+			// refresh auth bearer token if it expired
+			refreshUserAUthCtx(&config, clienterrors.IsUnauthorizedError, err)
+
+			return true, errors.Wrapf(err, "Unable to get Tanzu Mission Control nodepool entry, name : %s", d.Get(nodePoolNameKey))
+		}
+		// always run
+		d.SetId(resp.Nodepool.FullName.Name + ":" + resp.Nodepool.FullName.ClusterName)
+
+		return false, nil
+	}
+	timeoutValueData, _ := d.Get(waitKey).(string)
+
+	if ctx.Value(contextMethodKey{}) != "create" {
```

Comment:
can you revisit this logic?
I believe, is this needed. 

what is the advantage of this if check?

Reply:
That was a miss. Removed it. Calling the `getNodepoolResourceRetryableFn` below.